src/lsdb/dask/merge_catalog_functions.py

```diff
@@ -18,15 +18,52 @@
     from lsdb.catalog.catalog import Catalog
 
 
+def align_catalogs(left: Catalog, right: Catalog) -> PixelAlignment:
+    """Aligns two catalogs, also using the right catalog's margin if it exists
+
+    Args:
+        left (Catalog): The left catalog to align
+        right (Catalog): The right catalog to align
+    Returns:
+        The PixelAlignment object from aligning the catalogs
+    """
+
+    if right.margin is not None:
+        right_tree = align_trees(
+            right.hc_structure.pixel_tree,
+            right.margin.hc_structure.pixel_tree,
+            alignment_type=PixelAlignmentType.OUTER,
+        ).pixel_tree
+    else:
+        right_tree = right.hc_structure.pixel_tree
+    return align_trees(left.hc_structure.pixel_tree, right_tree, alignment_type=PixelAlignmentType.INNER)
+
+
+def join_partition_and_margin(df: pd.DataFrame, margin_df: pd.DataFrame, columns: List[str]) -> pd.DataFrame:
```

Comment:
This feels like a bit of an extreme to extract into a helper method. Do you anticipate that this would be called in many places? Could it just be inline instead?

Reply:
Yeah, that's fair. It'll be used in the join methods, but it doesn't really need a method. I'll inline it